cmd/gcp-controller-manager/main.go

```diff
@@ -75,6 +75,7 @@ var (
 	delayDirectPathGSARemove           = pflag.Bool("delay-direct-path-gsa-remove", false, "Delay removal of deleted Direct Path workloads' Google Service Accounts.")
 	hmsAuthorizeSAMappingURL           = pflag.String("hms-authorize-sa-mapping-url", "", "URL for reaching the Hosted Master Service AuthorizeSAMapping API.")
 	hmsSyncNodeURL                     = pflag.String("hms-sync-node-url", "", "URL for reaching the Hosted Master Service SyncNode API.")
+	csrAllowListPath                   = pflag.String("csr-allow-list-path", "", "An allow list for csr auto approve, we will put username in the allow list")
```

Comment:
instead of reading a file path could this instead be passed as a string config or and env? I really don't like the idea of adding files.

Reply:
I just afraid that the list will be too long, so pass it as a file

Reply:
can you list some examples?

Reply:
we have the metrics
```
kubelet_http_requests_total{long_running="false",method="GET",path="healthz",server_type="readonly"} 1
kubelet_http_requests_total{long_running="false",method="GET",path="metrics",server_type="readonly"} 31
kubelet_http_requests_total{long_running="false",method="GET",path="metrics/probes",server_type="readonly"} 23
kubelet_http_requests_total{long_running="false",method="GET",path="pods",server_type="readonly"} 4
kubelet_http_requests_total{long_running="false",method="GET",path="stats",server_type="readonly"} 69
```
anyone who is using those port will be migrate to allow list

for example, Datadog and Splunk

Reply:
Im not sure what example Datadog and Splunk would look like in the file. How is the datastructured in that file?

Reply:
There is a username in the csr, we will compare it with allow list. if the user name is datadog or splunk. we will auto approve the csr request